rubrics/impl/src/main/java/org/sakaiproject/rubrics/logic/RubricsServiceImpl.java

```diff
@@ -787,34 +791,51 @@ public void deleteRubricAssociationsByItemIdPrefix(String itemId, String toolId)
 					parameters).toObject(resourceParameterizedTypeReference);
 
 			for (Resource<ToolItemRubricAssociation> associationResource : associationResources) {
-				deleteRubricAssociation(associationResource.getLink(Link.REL_SELF).getHref(), toolId, null);
+				String associationHref = associationResource.getLink(Link.REL_SELF).getHref();
+				String [] assocSplitted = associationHref.split("/");
+				deleteRubricEvaluationsForAssociation(assocSplitted[assocSplitted.length-1], toolId);
+				deleteRubricElement(associationHref, toolId, null);
 			}
         } catch (Exception e) {
             log.warn("Error deleting rubric association for id {} : {}", itemId, e.getMessage());
         }
     }
-	
+
+    public void deleteRubricEvaluationsForAssociation(String associationId, String tool){
+        try{
+            log.debug("Deleting evaluations for association {}", associationId);
+            Collection<Resource<Evaluation>> evaluations = getRubricEvaluationsByAssociation(Long.valueOf(associationId));
+            for(Resource<Evaluation> eval : evaluations){
+                deleteRubricElement(eval.getLink(Link.REL_SELF).getHref(), tool, null);
+            }
+        } catch (Exception e) {
+            log.warn("Error deleting rubric association for tool {} and association {} : {}", tool, associationId, e.getMessage());
+        }
+	}
+
     public void deleteRubricAssociation(String tool, String id){
         try{
             Optional<Resource<ToolItemRubricAssociation>> associationResource = getRubricAssociationResource(tool, id, null);
             if (associationResource.isPresent()) {
-                deleteRubricAssociation(associationResource.get().getLink(Link.REL_SELF).getHref(), tool, null);
+                String associationHref = associationResource.get().getLink(Link.REL_SELF).getHref();
+                String [] assocSplitted = associationHref.split("/");
+                deleteRubricEvaluationsForAssociation(assocSplitted[assocSplitted.length-1], tool);
+                deleteRubricElement(associationHref, tool, null);
             }
         } catch (Exception e) {
             log.warn("Error deleting rubric association for tool {} and id {} : {}", tool, id, e.getMessage());
         }
     }
 
     /**
-     * Delete the rubric association.
-     * @param query The association href.
+     * Delete a rubric element (rubric, association).
+     * @param query The element href.
      * @return
      */
-
-    private void deleteRubricAssociation(String query, String toolId, String siteId) throws IOException {
-
+    private void deleteRubricElement(String query, String toolId, String siteId) throws IOException {
         HttpURLConnection conn = null;
```

Comment:
I'd change the name of this method from deleteRubricElement. It sounds like an HTML element of some kind and may be confusing later when we're in a project with web components. Maybe deleteRubricItem, or even deleteRubricThing ? No big deal, just my thought.

Reply:
What about deleteRubricResource?